pkg/apis/monitoring/v1/thanos_types.go

```diff
@@ -185,6 +185,12 @@ type ThanosRulerSpec struct {
 	// of all alerts.
 	// Maps to the '--alert.query-url' CLI arg.
 	AlertQueryURL string `json:"alertQueryUrl,omitempty"`
+	// Minimum number of seconds for which a newly created pod should be ready
+	// without any of its container crashing for it to be considered available.
+	// Defaults to 0 (pod will be considered available as soon as it is ready)
+	// This is an alpha field and requires enabling StatefulSetMinReadySeconds feature gate.
+	// +optional
+	MinReadySeconds uint32 `json:"minReadySeconds,omitempty"`
```

Comment:
I wonder if it is better to use `*uint32`. In this particular case it shouldn't matter too much since the default is 0 but if we are to ever change the default to say `5` then it will help with determining if the value was explicitly set by the user to `0` or if it was left empty (nil).

Reply:
On second thought, I agree with you. Sorry @PhilipGough for the back and forth :)

Reply:
no worries